x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -431,10 +432,16 @@ else if (plan instanceof OrderBy) {
                 }
             }
 
-            return result;
+            return filterUnsupportedProjections(result);
         }
 
-        static List<NamedExpression> expandStar(UnresolvedStar us, List<Attribute> output) {
+        private List<NamedExpression> filterUnsupportedProjections(List<NamedExpression> projections) {
```

Comment:
@bpintea Since this is your change, could you please explain? I also don't get this change.

Reply:
> Expanding star already filters out unsupported types

It's for this purpose, but I realise there's unfortunately a naming clash here. The "unsupported" in `filterUnsupportedProjections` stems from client-unsupported, not SQL-unsupported (i.e. DataType == UNSUPPORTED).
  
> @bpintea Since this is your change, could you please explain? I also don't get this change

This change has this [discussion](https://github.com/elastic/elasticsearch/pull/65145/files#r535223338) attached, which hasn't been concluded (and sorry for maybe not making clear this about the [stub PR](https://github.com/elastic/elasticsearch/pull/67433)).
The TLDR of that discussion: the star expansion of `SELECT * ...` will (potentially) include new fields of types that aren't compatible with old clients, so we need to hide them (both from catalog discovery and `SELECT *`).